### PR TITLE
[skyy] Change Server Loading

### PR DESCRIPTION
**Pull request by pbstFusion**
Change it to sky's loading sequence. Why? Because yes.

### CODE REVIEW

File: Loader/Loader/Loader.server.lua [hunk 1]
```diff
@@ -9,88 +9,130 @@
...
---	                  This is not designed to work in solo mode                       --
+--		This is not designed to work in solo mode.				--
 ----------------------------------------------------------------------------------------
 
-warn(":: Adonis :: Loading...");
+local print = function(...) for i,v in pairs({...}) do warn(":: Adonis ServerLoader :: INFO: "..tostring(v)) end end
+local error = function(...) for i,v in pairs({...}) do warn(":: Adonis ServerLoader :: ERROR: "..tostring(v).."; Traceback:\n"..debug.traceback()) end end
+local warn = function(...) for i,v in pairs({...}) do warn(":: Adonis ServerLoader:: WARN: "..tostring(v)) end end
+local pcall = function(func, ...) local ran, rerror = pcall(func, ...) if not ran then error(rerror) end return ran, rerror end
+local AbortLoad = function(Reason) warn("Adonis aborted loading. Reason: "..tostring(Reason)) if script then script:Destroy() end return false end
```
Comment (mewwo) [inline Loader/Loader/Loader.server.lua:25]: There is no reason to call `tostring` inside of `warn`, as all output functions call it already naturally.
This is why this happens when you print any `Instance`.
```lua
warn(workspace) --> Workspace (equals to workspace.Name)
```

You could improve the `warn` output by utilizing `string.format` to make it more readable.
Here is my suggested approach to making your code more clear and functional.
```suggestion
local function outputMessage(outputType, outputMessage)
    return warn(string.format(":: Adonis ServerLoader :: %s: %s"), outputType, outputMessage)
end

local function pcall(testFunction, ...)
    local success, returnValue = pcall(testFunction, ...)

    if not success then
        return outputMessage("ERROR", returnValue)
    end

    return success, returnValue
end

local function initiateAbort(abortReason)
    outputMessage("ABORTING", string.format("With the reason of %s", abortReason))
    return script:Destroy()
end
```

There is no reason for the `AbortLoad` function to have any return. My suggested change's code is still really unoptimized and a dumb idea overall, but it is definitely better whatever you wrote.

There is no overall improvement in your pull request, it is just randomly changed naming styles for few variables.

File: Loader/Loader/Loader.server.lua [hunk 2]
```diff
@@ -9,88 +9,130 @@
...
-	local a,setTab = pcall(require,settings)
-	if not a then
-		warn'::Adonis:: Settings module errored while loading; Using defaults;'
-		setTab = {}
+	local ModuleID = Data.ModuleID
+	local success, settings = pcall(require, Settings)
+
+	if not success then
+		warn("Settings module failed to load; Using defaults;")
+		settings = {}
```
Comment (mewwo) [inline Loader/Loader/Loader.server.lua:111]: You have just reimplemented `pcall` to output an error message if the function has failed, yet you have decided to output a `warn` manually?

**Review by mewwo — COMMENTED** [review]
Feel like most of this pull request is pointless and serves no functional purpose except changed variable names.

File: Loader/Loader/Loader.server.lua (hunk 2 above)
Comment (pbstFusion) [reply Loader/Loader/Loader.server.lua:124]: Can you resubmit this, i can't change it because it's "outdated"

**Comment by pbstFusion** [comment]
<@&431593291776131082>

**Comment by pbstFusion** [comment]
ok good

File: Loader/Loader/Loader.server.lua [hunk 3]
```diff
@@ -9,88 +9,143 @@
...
+        return outputMessage("ERROR", returnValue)
+    end
+
+    return success, returnValue
+end
 
-if _G["__Adonis_MUTEX"] and type(_G["__Adonis_MUTEX"])=="string" then
-	warn("\n-----------------------------------------------"
+local function initiateAbort(abortReason)
+    outputMessage("ABORTING", string.format("With the reason of %s", abortReason))
```
Comment (ccuser44) [inline Loader/Loader/Loader.server.lua:36]: "With the reason of " sounds a bit odd maybe "Reason: %s" sounds better

File: Loader/Loader/Loader.server.lua [hunk 4]
```diff
@@ -9,88 +9,143 @@
...
+--		Edit settings in-game or using the settings module in the Config folder	--
 ----------------------------------------------------------------------------------------
---	                  This is not designed to work in solo mode                       --
+--		This is not designed to work in solo mode.				--
 ----------------------------------------------------------------------------------------
 
-warn(":: Adonis :: Loading...");
+local function outputMessage(outputType, outputMessage)
+    return warn(string.format(":: Adonis ServerLoader :: %s: %s"), outputType, outputMessage)
+end
```
Comment (ccuser44) [inline Loader/Loader/Loader.server.lua:23]: Replacing warn is fine. I don't think a new function outputMessage should be added
You should instead do
`
local OldWarn = warn
local warn = function(...) OldWarn(:: Adonis ServerLoader:: WARN: ", ...) end
`

File: Loader/Loader/Loader.server.lua [hunk 5]
```diff
@@ -9,88 +9,143 @@
...
+--		Epix Incorporated. Not Everything is so Black and White.		--
 ----------------------------------------------------------------------------------------
---	    Edit settings in-game or using the settings module in the Config folder	      --
+--		Edit settings in-game or using the settings module in the Config folder	--
 ----------------------------------------------------------------------------------------
---	                  This is not designed to work in solo mode                       --
+--		This is not designed to work in solo mode.				--
 ----------------------------------------------------------------------------------------
 
-warn(":: Adonis :: Loading...");
```
[the remark this answers is not in the document]
Comment (pbstFusion) [reply Loader/Loader/Loader.server.lua:21]: because it just uses console log space just for the loader to say "i am working"

**Comment by pbstFusion** [comment]
ok

**Comment by GalacticInspired** [comment]
ok

File: Loader/Loader/Loader.server.lua (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sceleratis) [reply Loader/Loader/Loader.server.lua:21]: "I am working" is something very important to know when something seems to not be working and needs debugging

File: Loader/Loader/Loader.server.lua (hunk 4 above)
Comment (Sceleratis) [reply Loader/Loader/Loader.server.lua:23]: Adonis already replaces warn and print

**Comment by Sceleratis** [comment]
Aside from some issues, skyy said he's redoing his original pull for the current build.

**Comment by Sceleratis** [comment]
skyy has been redoing/resubmitting stuff and this has issues so I'm gonna close it

**Pull request closed without merging**